diff --git a/include/envoy/server/drain_manager.h b/include/envoy/server/drain_manager.h
index d9398ee..61a810a 100644
--- a/include/envoy/server/drain_manager.h
+++ b/include/envoy/server/drain_manager.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <memory>
+#include <functional>
 
 #include "envoy/network/drain_decision.h"
 
diff --git a/include/envoy/server/worker.h b/include/envoy/server/worker.h
index d52a181..60490a3 100644
--- a/include/envoy/server/worker.h
+++ b/include/envoy/server/worker.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <functional>
+
 #include "envoy/server/guarddog.h"
 
 namespace Envoy {
diff --git a/source/common/common/callback_impl.h b/source/common/common/callback_impl.h
index f725d80..4172d34 100644
--- a/source/common/common/callback_impl.h
+++ b/source/common/common/callback_impl.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <list>
+#include <functional>
 
 #include "envoy/common/callback.h"
 
diff --git a/source/common/config/subscription_factory.h b/source/common/config/subscription_factory.h
index 63dbe93..a394a98 100644
--- a/source/common/config/subscription_factory.h
+++ b/source/common/config/subscription_factory.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <functional>
+
 #include "envoy/config/subscription.h"
 
 #include "common/config/filesystem_subscription_impl.h"
diff --git a/source/server/drain_manager_impl.cc b/source/server/drain_manager_impl.cc
index f5fe250..6cd5226 100644
--- a/source/server/drain_manager_impl.cc
+++ b/source/server/drain_manager_impl.cc
@@ -2,6 +2,7 @@
 
 #include <chrono>
 #include <cstdint>
+#include <functional>
 
 #include "envoy/event/dispatcher.h"
 #include "envoy/event/timer.h"
diff --git a/source/server/init_manager_impl.cc b/source/server/init_manager_impl.cc
index 14c963e..e20c3af 100644
--- a/source/server/init_manager_impl.cc
+++ b/source/server/init_manager_impl.cc
@@ -1,3 +1,5 @@
+#include <functional>
+
 #include "server/init_manager_impl.h"
 
 #include "common/common/assert.h"
diff --git a/source/server/lds_api.cc b/source/server/lds_api.cc
index 40d9eeb..3c8519e 100644
--- a/source/server/lds_api.cc
+++ b/source/server/lds_api.cc
@@ -1,3 +1,5 @@
+#include <functional>
+
 #include "server/lds_api.h"
 
 #include "common/config/utility.h"
diff --git a/source/server/lds_api.h b/source/server/lds_api.h
index d8cb499..6998592 100644
--- a/source/server/lds_api.h
+++ b/source/server/lds_api.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <functional>
+
 #include "envoy/init/init.h"
 #include "envoy/server/listener_manager.h"
 #include "envoy/stats/stats_macros.h"
diff --git a/source/server/worker_impl.cc b/source/server/worker_impl.cc
index a0a4e23..9ff1944 100644
--- a/source/server/worker_impl.cc
+++ b/source/server/worker_impl.cc
@@ -1,3 +1,5 @@
+#include <functional>
+
 #include "server/worker_impl.h"
 
 #include "envoy/event/dispatcher.h"
diff --git a/source/server/worker_impl.h b/source/server/worker_impl.h
index 749f5bb..4922666 100644
--- a/source/server/worker_impl.h
+++ b/source/server/worker_impl.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <memory>
+#include <functional>
 
 #include "envoy/api/api.h"
 #include "envoy/network/connection_handler.h"
diff --git a/test/common/grpc/json_transcoder_filter_test.cc b/test/common/grpc/json_transcoder_filter_test.cc
index 914c489..a3672bb 100644
--- a/test/common/grpc/json_transcoder_filter_test.cc
+++ b/test/common/grpc/json_transcoder_filter_test.cc
@@ -1,4 +1,5 @@
 #include <fstream>
+#include <functional>
 
 #include "common/buffer/buffer_impl.h"
 #include "common/filesystem/filesystem_impl.h"
